pkg/action/action.go

```diff
@@ -422,6 +427,12 @@ func (cfg *Configuration) Init(getter genericclioptions.RESTClientGetter, namesp
 	cfg.KubeClient = kc
 	cfg.Releases = store
 	cfg.Log = log
+	cfg.HookOutputFunc = defaultHookOutputWriter
```

Comment:
> Additionally, there will be a new item added to the action SDK configuration to allow SDK consumers to get the output. By default this output will be discarded by writing to io.Discard, but an SDK consumer can overwrite the HookOutputFunc to provide a custom writer.

Snippet from the HIP.

We need to allow SDK users to provide their own `HookOutputFunc`. The default would be one that writes to `io.Discard`

Reply:
I made changes to the Init method signature before I saw your second comment. 

How do SDK users normally interact with the `Configuration` struct? I assumed they would build their own rather than use the `Init`...

Assuming we can't change the `Init` signature and we can't default to a function returning `iog.Writer()` then the only option is to set the `cfg.HookOutputFunc` after the `Init` func has run. I can add a utility to do this for now, but if you have a preferred alternative please let me know.

Reply:
https://helm.sh/docs/sdk/examples/ will have examples of how an SDK user would use Helm

Reply:
We should be able to change the public interfaces, be we cannot backport this PR to v3 helm.

Reply:
Ok - given this PR no longer changes the existing interfaces I think we should be good to have this in v3. 

Looking at the SDK examples I think the current solution of setting the `HookOutputFunc` in the `main` function after the `Init` has run should work.

Reply:
> I can add a utility to do this for now, but if you have a preferred alternative please let me know.

This could work as wel.

Reply:
@Bez625 Nice job 👍 This works! Thanks for @banjoh for clarifying @gjenkins8 's comment https://github.com/helm/helm/pull/10309#discussion_r1938374610 that it needed to be user-configurable. It is now, so I think we are good on this.